### PR TITLE
Support CYW43 boards on RP2350

### LINKED ISSUE #2608: Support multiple CYW43 driver configurations for RP2350
I did some investigation and there are several things that need to fall into place to get CYW43 running on non-RP2040-PicoW boards.

`libpico` needs to be updated on the RP2350 builds to set the proper config and libraries.  Right now only the RP2040 will include the cyw43 driver stack.  `tools/libpico/Cmakelists.txt` and rebuild, which leads to the first issue.

The 2.0.0 SDK has hardcoded defines for the pinout of the CYW43 chip.  On the RP2040/PicoW that's no big deal, there is only one physical setup to worry about.  On the RP2350 with multiple Wifi PCBs, that will need to be abstracted (we need to build the SDK using Cmake outside of Arduino, so only build a single .A for all CPU archs, not for all PCBs).  Defines like `CYW43_PIN_WL_HOST_WAKE` and `CYW43_PIN_WL_SDIO_1`, etc.  This can probably be handled with some careful `cores/rp2040/sdkoverrides` but I'd need to think about how to safely implement it...

Once that's running, then an additional define, `PICO_CYW43_SUPPORTED` needs to be added to the `extra` list in `tools/makeboards.py` for the PicoW and RP2350 boards w/the CYW43 chip.  Maybe including the physical `CYW43_PIN_xxxx` from above.  And then an almost search/replace of `s/ifdef ARDUINO_RASPBERRY_PI_PICO_W/ifdef PICO_CYW43_SUPPORTED/` through the libraries and you're within spitting distance...

So, I see a couple options.  You may want to consider going radio-free to start, and then I can work on my own to get the CYW43 stuff up later.  Or, I can start a new dev branch/PR with the changes above and you can rebase this PR to that branch.  When it's good, we can merge this Pr into that branch, then merge that branch to `master`.  Whatever makes sense for you, just let me know...

--edit-- One other option would be to split out the cyw43 Cmake into its own library and pull that in on an individual basis.  If the constant definitions become too much of a hangup, then a hardcoded list of pinouts in the `CMakefile.txt` could be built and then linked in via `makeboards.py` adding the appropriate flags.

_Originally posted by @earlephilhower in https://github.com/earlephilhower/arduino-pico/issues/2605#issuecomment-2476743384_

### PR DESCRIPTION
**Pull request by earlephilhower**
Using pico-sdk develop branch with minor patch, add in support for the CYW43-based WiFi boards on the RP2350.

Fixes #2608

### CODE REVIEW

**Comment by earlephilhower** [comment]
FWIW, this should now build on the RP2350+CYW43 boards (Pimoroni PicoPlus 2W and SparkFun Thing Plus RP2350) while also not breaking PicoW.  SparkFun was nice enough to send a sample, so once that arrives final tweaks can be done.

On the bright side, it appears all boards are using the same CYW43 pin out.  The SDK should allow for changing it per-board, but as of now there doesn't seem to be a need to exercise this.

This also pulls in #2600 since it is trivial and part of the refactoring done to support multiple CYW43 boards.

SDK builds in CI will fail until the next SDK release (and https://github.com/raspberrypi/pico-sdk/pull/2045 gets merged).

File: cores/rp2040/cyw43_wrappers.cpp [hunk 1]
```diff
@@ -101,4 +107,63 @@ extern "C" void __wrap_cyw43_cb_tcpip_deinit(cyw43_t *self, int itf) {
...
+extern "C" void init_cyw43_wifi() {
+    __isPicoW = CheckPicoW();
+    if (__isPicoW) {
+        // Fix for overclocked CPU: SPI communication breaks down with default "div by 2" speed
+        // So, divide clock by 4 for anything including and above 250MHz CPU frequency.
+        if (clock_get_hz(clk_sys) >= 250000000) {
+            cyw43_set_pio_clock_divisor(4, 0); // div by 4.0
+        }
+        cyw43_arch_init_with_country(WIFICC);
+    }
```
[the remark this answers is not in the document]
Comment (earlephilhower) [reply cores/rp2040/cyw43_wrappers.cpp:158]: The check is ifdef'd out on non-RP2040 builds and always returns true there.

**Comment by sfe-SparkFro** [comment]
> On the bright side, it appears all boards are using the same CYW43 pin out. The SDK should allow for changing it per-board, but as of now there doesn't seem to be a need to exercise this.

I haven't poked through the code changes, but does this statement imply there's no way to use a different radio pinout with this PR? I would request that to be a feature, because we have another product in the works that will use different pins (can't say anything more at this time).

**Comment by earlephilhower** [comment]
> I haven't poked through the code changes, but does this statement imply there's no way to use a different radio pinout with this PR? I would request that to be a feature, because we have another product in the works that will use different pins (can't say anything more at this time).

No worries, it'll be configurable at runtime.  `CYW43_PIN_WL_DYNAMIC=1` in the `CMakefile.txt` *should* allow a calls to `cyw43_hal_pin_config()` in the board variant initializer before the `cyw43_arch_init()` call.

The default values, though, if the pins aren't redefined by calls to the pin_config process, are all the same as PicoW and the SparkFun and Pimoroni boards.  So, as of now, there's no need to set anything and it "should" pick up the proper defaults everywhere.

**Comment by sfe-SparkFro** [comment]
Ok, I had a play around with this PR and I got it working with different pins.

Turns out `cyw43_set_pins_wl()` is what's needed, not `cyw43_hal_pin_config()`. I put the following in my `variant/init.cpp`:

```
#include <cyw43_wrappers.h>
#include "pico/cyw43_driver.h" // Needed for cyw43_set_pins_wl()

extern "C" void initVariant() {
    static uint cyw43_pin_array[CYW43_PIN_INDEX_WL_COUNT] = {0, 1, 2, 3, 4, 5};
    cyw43_set_pins_wl(cyw43_pin_array);
    init_cyw43_wifi();
}
```

where `{0, 1, 2, 3, 4, 5}` is a placeholder for the actual radio pins.

The only major problem is that `CYW43_PIN_WL_DYNAMIC` defaults to `0`, so `cyw43_set_pins_wl()` is not available to call. As a workaround, I just added `#define CYW43_PIN_WL_DYNAMIC 1` to my  `variant/pins_arduino.h`, though that's arguably not the right place for it.

**Comment by earlephilhower** [comment]
>  ...I just added `#define CYW43_PIN_WL_DYNAMIC`...

It's set in the library build but should also be passed in as part of the `makeboards` extra flags so the IDE and P.IO will both get it.  It'll be in the next push.

**Comment by maxgerhardt** [comment]
Awesome work!! I've just updated the PlatformIO integration and I can run the WiFi scan example on my Pimoroni Pico 2 W perfectly fine!
```
Beginning scan at 5001
Found 5 networks

                            SSID   ENC     BSSID         CH RSSI
                Vodafone Hotspot  NONE 72:54:25:XX:XX:XX  4  -65
               Vodafone Homespot  NONE 72:54:25:XX:XX:XX  4  -69
...
```

**Comment by earlephilhower** [comment]
Awesome, thanks for the report!  I just used the schematic and they seemed to have chosen the exact same pinout as all the others.

Now to figure out what to do with a WiFi-enabled, 512K, dual core with FP board. :smile:

**Pull request merged**